code/modules/reagents/reagent_containers/food/snacks.dm

```diff
@@ -92,8 +96,27 @@
 		consume(user, 1)
 
 /obj/item/weapon/reagent_containers/food/snacks/New()
-
 	..()
+	if(food_flags & FOOD_NEEDS_COLD)
+		processing_objects.Add(src)
+
+/obj/item/weapon/reagent_containers/food/snacks/proc/Rot()
+	rotten = 1
+	return
+
+/obj/item/weapon/reagent_containers/food/snacks/process()
+	if(istype(loc, /obj/structure/closet/secure_closet/freezer))
+		return
+	cold--
+
+	if(cold <= 0)
+		var/area/A = get_area(src.loc)
+		Rot()
+		cold = 0
+		name = "rotten [src.name]"
+		if(prob(100))
```

Comment:
I assume, based on your description in the OP, that this was set to 100 for testing purposes and will be changed to "a low chance"

Don't forget